Rahul/ProxyPattern/test.cpp

```diff
@@ -0,0 +1,26 @@
+#include "AbstractOperation.hpp"
+#include "ProxyAstronomicOperation.hpp"
+#include <iostream>
+
+int main(){
+    AbstractOperationPtr proxy = AbstractOperationPtr( (AbstractOperation *)
```

Comment:
minor thing..I think you don't really need to do manual casting...upcast is implicit